addons/mrp/models/mrp_bom.py

```diff
@@ -438,19 +429,10 @@ class MrpByProduct(models.Model):
         'mrp.routing.workcenter', 'Produced in Operation', check_company=True,
         domain="[('id', 'in', allowed_operation_ids)]")
 
-    @api.depends('bom_id')
+    @api.depends('bom_id.operation_ids')
     def _compute_allowed_operation_ids(self):
         for byproduct in self:
-            if not byproduct.bom_id.operation_ids:
-                byproduct.allowed_operation_ids = self.env['mrp.routing.workcenter']
-            else:
-                operation_domain = [
-                    ('id', 'in', byproduct.bom_id.operation_ids.ids),
-                    '|',
-                        ('company_id', '=', byproduct.company_id.id),
-                        ('company_id', '=', False)
-                ]
-                byproduct.allowed_operation_ids = self.env['mrp.routing.workcenter'].search(operation_domain)
+            byproduct.allowed_operation_ids = byproduct.bom_id.operation_ids.ids
```

Comment:
looks like a related

Reply:
Yes, agree. I just need to change the type of fields in One2many. Thanks, it cleaner.